src/geom/sides.rs

```diff
@@ -266,3 +371,25 @@ impl<T: Fold> Fold for Sides<Option<T>> {
         })
     }
 }
+
+impl Fold for Margin {
+    type Output = Margin;
+
+    fn fold(self, outer: Self::Output) -> Self::Output {
+        let sides =
+            self.sides
+                .zip(outer.sides)
+                .map(|(inner, outer)| match (inner, outer) {
+                    (Some(value), Some(outer)) => Some(value.fold(outer)),
+                    (Some(value), None) => Some(value),
+                    (None, Some(outer)) => Some(outer),
+                    (None, None) => None,
+                });
+        let two_sided = match (self.two_sided, outer.two_sided) {
+            (_, Some(two_sided)) => Some(two_sided),
+            (Some(two_sided), None) => Some(two_sided),
+            (None, None) => None,
+        };
```

Comment:
Here, you're giving fold priority to `outer`, which isn't consistent with other types, such as stroke. https://github.com/typst/typst/blob/4e4413e7beb378f02f2f2f76abde125ecdbd3c69/src/geom/stroke.rs#L194

Instead, give priority to inner (`self`). (Also, simplify.)
```suggestion
        let two_sided = self.two_sided.or(outer.two_sided);
```

Reply:
Done.